src/mc.rs

```diff
@@ -2300,6 +2300,70 @@ impl Rav1dMCDSPContext {
         self.warp8x8t = bd_fn!(warp8x8t::decl_fn, BD, warp_affine_8x8t, neon);
         self.emu_edge = bd_fn!(emu_edge::decl_fn, BD, emu_edge, neon);
 
+        #[cfg(target_feature = "dotprod")]
+        if BD::BITDEPTH == 8 {
+            if !flags.contains(CpuFlags::DOTPROD) {
+                return self;
+            }
+
+            self.mc = enum_map!(Filter2d => mc::Fn; match key {
+                Regular8Tap => bd_fn!(mc::decl_fn, BD, put_8tap_regular, neon_dotprod),
```

Comment:
I think you need to use `bpc_fn!` to declare only the 8-bitdepth version, otherwise it declares both (same for all the rest)

```suggestion
                Regular8Tap => bpc_fn!(mc::decl_fn, 8 bpc, put_8tap_regular, neon_dotprod),
```

Reply:
Good point. I made the change